src/Settings/GalaSettings.vala

```diff
@@ -58,9 +58,14 @@ namespace ElementaryTweaks {
         static Gee.HashMap<string, string> preset_button_layouts;
         static Gtk.ListStore button_layouts;
         static AppearanceSettings? instance = null;
+        static GLib.Settings gnome_wm_settings = new GLib.Settings ("org.gnome.desktop.wm.preferences");
 
         private AppearanceSettings () {
             base (SCHEMA + ".appearance");
+
+            changed.connect (() => {
+                gnome_wm_settings.set_string ("button-layout", button_layout);
```

Comment:
As an FYI for future changes, this would have updated the `gnome_wm_settings` after any property changes in the AppearanceSettings class. In this case it is not impacting now, but thought you should know :)

Reply:
Thank you for your quick review and comments!

Do you think [my first solution](https://github.com/elementary-tweaks/elementary-tweaks/pull/124/commits/4e91551f36fdbb6bd4fd884db041c7f58b6629a1) would be better then? Because that one only updates `button-layout` only when value of `controls_combobox` is changed.

Reply:
I think that this one is better since it will keep all the settings backend in one place :)

Reply:
@Philip-Scott Do you think the following one would be a solution? :thinking:

```diff
-            changed.connect (() => {
-                gnome_wm_settings.set_string ("button-layout", button_layout);
-            });
+            gnome_wm_settings.bind ("button-layout", this, "button-layout", SettingsBindFlags.DEFAULT);
```

I hope this would only check the change of the value of property `button_layout`.